core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -286,6 +291,18 @@ public void validate() {
         throw new IllegalArgumentException("Port cannot be negative");
       }
 
+      if ("password".equalsIgnoreCase(authMechanism) && Strings.isNullOrEmpty(oracleServerPassword)) {
+        throw new IllegalArgumentException("Password cannot be empty");
+      } else if ("Private Key".equalsIgnoreCase(authMechanism) && Strings.isNullOrEmpty(privateKey)) {
+        throw new IllegalArgumentException("Private Key cannot be empty");
+      }
+
+      if (!(authMechanism.equalsIgnoreCase("Password") || authMechanism.equalsIgnoreCase("Private Key"))) {
```

Comment:
This condition should be moved even before line 294. also change this to use the string literal first in the equals condition.